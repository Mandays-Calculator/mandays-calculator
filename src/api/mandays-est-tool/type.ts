export interface TasksResponse {
  id: string;
  name: string;
  description: string;
  status: string;
  functionality: Functionality;
  tags: string;
  comments: Comments[];
  sprint: string;
  completionDate: string;
  createdDate: string;
}

interface Functionality {
  id: string;
  name: string;
}

interface Comments {
  id: string;
  description: string;
}
