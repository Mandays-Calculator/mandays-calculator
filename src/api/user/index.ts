export { getUsers } from "./User";
export * from "./types";
