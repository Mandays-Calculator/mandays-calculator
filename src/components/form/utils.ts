import type { FormErrors } from "./types";
import { defaultTo, get } from "lodash";

export const getOption = <Type extends object>(
  path: keyof Type | "value" | "label",
): ((option: Type) => string) => {
  return (option: Type) => {
    return defaultTo(get(option, path), "");
  };
};

export const getFieldError = (
  errors: FormErrors | undefined,
  name: string,
): string | undefined => {
  if (errors) {
    return typeof get(errors, name) === "string" ? get(errors, name) : "";
  }
};
