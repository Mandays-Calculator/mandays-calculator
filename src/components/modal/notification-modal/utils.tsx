import { ReactElement } from "react";
import { ModalType } from "./types";

import Grid from "@mui/material/Grid";
import WarningIcon from "@mui/icons-material/Warning";
import Typography, { TypographyProps } from "@mui/material/Typography";

import { SvgIcon } from "~/components";

interface IdleTimerSectionProps {
  idleTimer: number;
  idleTimerSeconds: number;
}

const StyledTypography: React.FC<TypographyProps> = (
  props: TypographyProps
): ReactElement => (
  <Typography
    variant="inherit"
    fontSize="3.5rem"
    fontFamily="sans-serif"
    {...props}
  />
);

export const IdleTimerSection: React.FC<IdleTimerSectionProps> = ({
  idleTimer,
  idleTimerSeconds,
}: IdleTimerSectionProps): ReactElement => (
  <Grid container spacing={1}>
    <Grid item xs={1} marginX="7rem">
      <StyledTypography>{idleTimer}min</StyledTypography>
    </Grid>
    <Grid item>
      <StyledTypography>{idleTimerSeconds}secs</StyledTypography>
    </Grid>
  </Grid>
);

export const renderIcon = (type: ModalType): ReactElement => {
  switch (type) {
    case "error":
    case "warning":
    case "unauthorized":
      return <SvgIcon name="warning" $size={8} />;
    case "idleTimeOut":
      return <WarningIcon color="error" sx={{ fontSize: 40 }} />;
    case "success":
    default:
      return <WarningIcon color="success" />;
  }
};

export const TimerSection = ({
  idleTimer,
  idleTimerSeconds,
}: {
  idleTimer: number;
  idleTimerSeconds: number;
}): ReactElement => (
  <>
    <IdleTimerSection
      idleTimer={idleTimer}
      idleTimerSeconds={idleTimerSeconds}
    />
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Typography variant="h5" style={{ margin: "1rem" }}>
          Please click "Stay Logged On" to keep working; or click "Logout" to
          end your session now.
        </Typography>
      </Grid>
    </Grid>
  </>
);
