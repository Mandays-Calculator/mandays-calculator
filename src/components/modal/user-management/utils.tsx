export const genders = [
  {
    label: "FEMALE",
    value: "FEMALE",
  },
  {
    label: "MALE",
    value: "MALE",
  },
  {
    label: "NON-BINARY",
    value: "NON_BINARY",
  },
  {
    label: "PREFER NOT TO SAY",
    value: "PREFER_NOT_TO_SAY",
  },
];

export const rolesData = [
  {
    label: "System Admin",
    value: "ROLE_SYS_ADMIN",
  },
  {
    label: "Sprint Manager",
    value: "ROLE_SPRINT_MANAGER",
  },
  {
    label: "User",
    value: "ROLE_USER",
  },
];
