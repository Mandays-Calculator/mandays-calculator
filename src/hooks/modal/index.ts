export * from "./useDisabledBodyScroll";
export * from "./useUnauthorizedModal";
