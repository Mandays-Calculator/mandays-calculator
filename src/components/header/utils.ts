import { UserProject } from "~/api/auth";

export const getProjectOptions = (projects: UserProject[]): SelectObject[] => {
  return projects
    .filter((project: UserProject) => project.active)
    .map((project) => ({
      label: project.name,
      value: project.projectId,
    }));
};
