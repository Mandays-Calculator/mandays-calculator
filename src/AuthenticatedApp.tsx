import type { ReactElement } from "react";
import type { AppDispatch } from "~/redux/store";
import type { UserPermissionState } from "~/redux/reducers/user/types";

import { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useDispatch, useSelector } from "react-redux";
import { useAuth } from "react-oidc-context";

import { PageLoader, Layout, NotificationModal } from "~/components";
import LocalizationKey from "~/i18n/key";
import { Auth } from "~/pages/auth";

import { getUser } from "~/utils/oidc-utils";
import { getEnvConfig } from "~/utils/env-config";
import { ERROR_MESSAGES, LOCAL_STORAGE_ITEMS } from "~/utils/constants";
import {
  removeStateStorage,
  getItemStorage,
  removeStorageListener,
  addStorageListener,
} from "~/utils/storageHelper";

import AppRoutes from "~/routes/AppRoutes";
import axiosInit from "~/api/axios.config";

import { fetchUserPermission, selectUser } from "~/redux/reducers/user";

const AuthenticatedApp = (): ReactElement => {
  const auth = useAuth();
  const user = getUser();
  const config = getEnvConfig();
  const dispatch: AppDispatch = useDispatch();
  const { t } = useTranslation();

  const userState: UserPermissionState = useSelector(selectUser);
  const [showUnauthorizedModal, setShowUnauthorizedModal] =
    useState<boolean>(false);

  const { common } = LocalizationKey;

  useEffect(() => {
    /**
     * Initiate axios token once user is logged in
     */
    if (user) {
      axiosInit();
      dispatch(fetchUserPermission());
    }

    /**
     * Track changes in local storage items
     */
    const handleStorageChange = () => {
      const sessionState = getItemStorage(LOCAL_STORAGE_ITEMS.sessionState);
      setShowUnauthorizedModal(sessionState.unauthorized);
    };

    handleStorageChange();
    addStorageListener(handleStorageChange);
    return () => {
      setShowUnauthorizedModal(false);
      removeStorageListener(handleStorageChange);
    };
  }, [auth]);

  const renderAuth = (): ReactElement => {
    switch (auth.activeNavigator) {
      case "signinSilent":
        return (
          <PageLoader
            labelOnLoad={t(common.userManagement.authSignInLoading)}
          />
        );
      case "signoutRedirect":
        return (
          <PageLoader
            labelOnLoad={t(common.userManagement.authSignOutLoading)}
          />
        );
    }

    if (auth.isLoading || userState.loading) {
      return (
        <PageLoader
          labelOnLoad={t(common.userManagement.authPermissionLoading)}
        />
      );
    }

    if (!config.enableAuth) {
      return (
        <Layout>
          <AppRoutes />
        </Layout>
      );
    } else {
      if (auth.isAuthenticated) {
        return (
          <Layout>
            <AppRoutes />
          </Layout>
        );
      }
      return <Auth />;
    }
  };

  return (
    <>
      {renderAuth()}
      {showUnauthorizedModal && (
        <NotificationModal
          type="unauthorized"
          message={t(common.errorMessage.unauthorized)}
          open={showUnauthorizedModal}
          onConfirm={() => {
            removeStateStorage();
            auth.removeUser();
          }}
        />
      )}
      {userState.error !== null && (
        <NotificationModal
          type="error"
          message={ERROR_MESSAGES.permission}
          open={userState.error !== null}
        />
      )}
    </>
  );
};

export default AuthenticatedApp;
