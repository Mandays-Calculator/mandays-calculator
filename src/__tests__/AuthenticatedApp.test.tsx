import type { ConfigType } from "~/utils/env-config";

import { BrowserRouter } from "react-router-dom";
import { useDispatch } from 'react-redux';

import { render, waitFor } from "@testing-library/react";

import { checkUserAuthentication, useUserAuth } from "~/hooks/user";
import AuthenticatedApp from "~/AuthenticatedApp";
import { getEnvConfig } from "~/utils/env-config";
import { useErrorModals } from "~/hooks/modal";

import { cleanAllCallback } from "./pages/auth/utils/auth-utils";

jest.mock("~/api/axios.config");
jest.mock("~/utils/env-config");
jest.mock("~/hooks/modal");
jest.mock("~/hooks/user");
jest.mock("react-redux");

const mockGetEnvConfig = getEnvConfig as jest.MockedFunction<
  typeof getEnvConfig
>;
const mockCheckUserAuthentication = checkUserAuthentication as jest.MockedFunction<
  typeof checkUserAuthentication
>;
const mockUseUserAuth = useUserAuth as jest.MockedFunction<
  typeof useUserAuth
>;
const mockUseDispatch = useDispatch as jest.MockedFunction<
  typeof useDispatch
>;
const mockUseErrorModals = useErrorModals as jest.MockedFunction<
  typeof useErrorModals
>;

afterAll((done) => {
  cleanAllCallback(done);
});

describe("GIVEN AuthenticatedApp Component is called", () => {
  test("WHEN page is still in verification process, THEN it should render the page loader", async () => {
    const checkUserAuthenticationData = {
      status: false,
      mcUser: 'AUTHENTICATEDUSERTOKENSAMPLE',
    };
    const getEnvConfigData = {
      enableAuth: false,
      idleTimeoutConfig: {
        durationUntilPromptSeconds: 100
      }
    };
    const useUserAuthData = {
      state: {
        loading: true,
        isAuthenticated: false,
        permissions: [{
          displayName: "ROLE_SYS_ADMIN",
          icon: "icon_sample",
          path: "icon_path",
          roles: ["ROLE_SYS_ADMIN"],
          subMenuItems: null
        }]
      }
    }
    const useErrorModalsData = {
      showUnauthorizedModal: false,
      systemErrorModal: false,
      setSystemErrorModal: jest.fn()
    }

    const renderResult = renderAuthenticatedApp(getEnvConfigData, checkUserAuthenticationData, useUserAuthData, useErrorModalsData);

    if (!renderResult) {
      throw new Error("Rendering failed.");
    }

    const { container } = renderResult;

    await waitFor(() => {
      expect(container).toMatchSnapshot(`auth-page-loader`);
    });
  });

  test("WHEN user is not authenticated, THEN it user cannot proceed to the landing page", async () => {
    const checkUserAuthenticationData = {
      status: false,
      mcUser: 'AUTHENTICATEDUSERTOKENSAMPLE',
    };
    const getEnvConfigData = {
      enableAuth: false,
      idleTimeoutConfig: {
        durationUntilPromptSeconds: 100
      }
    };
    const useUserAuthData = {
      state: {
        loading: false,
        isAuthenticated: false,
        permissions: [{
          displayName: "ROLE_SYS_ADMIN",
          icon: "icon_sample",
          path: "icon_path",
          roles: ["ROLE_SYS_ADMIN"],
          subMenuItems: null
        }]
      }
    }
    const useErrorModalsData = {
      showUnauthorizedModal: false,
      systemErrorModal: false,
      setSystemErrorModal: jest.fn()
    }

    const renderResult = renderAuthenticatedApp(getEnvConfigData, checkUserAuthenticationData, useUserAuthData, useErrorModalsData);

    if (!renderResult) {
      throw new Error("Rendering failed.");
    }

    const { container } = renderResult;

    await waitFor(() => {
      expect(container).toMatchSnapshot(`user-not-authenticated`);
    });
  });

  test("WHEN user is authenticated, THEN it user should be redirected to the landing page", async () => {
    const checkUserAuthenticationData = {
      status: true,
      mcUser: 'AUTHENTICATEDUSERTOKENSAMPLE',
    };
    const getEnvConfigData = {
      enableAuth: true,
      idleTimeoutConfig: {
        durationUntilPromptSeconds: 100
      }
    };
    const useUserAuthData = {
      state: {
        loading: false,
        isAuthenticated: true,
        permissions: [{
          displayName: "ROLE_SYS_ADMIN",
          icon: "icon_sample",
          path: "icon_path",
          roles: ["ROLE_SYS_ADMIN"],
          subMenuItems: null
        }]
      }
    }
    const useErrorModalsData = {
      showUnauthorizedModal: false,
      systemErrorModal: false,
      setSystemErrorModal: jest.fn()
    }

    const renderResult = renderAuthenticatedApp(getEnvConfigData, checkUserAuthenticationData, useUserAuthData, useErrorModalsData);

    if (!renderResult) {
      throw new Error("Rendering failed.");
    }

    const { container } = renderResult;

    await waitFor(() => {
      expect(container).toMatchSnapshot(`user-authenticated`);
    });
  });
});

const renderAuthenticatedApp = (
  getEnvConfigData: object,
  checkUserAuthenticationData: object,
  useUserAuthData: object,
  useErrorModalsData: object
): ReturnType<typeof render> | undefined => {
  runAuthenticatedAppMocks(getEnvConfigData, checkUserAuthenticationData, useUserAuthData, useErrorModalsData);

  return render(
    <BrowserRouter>
      <AuthenticatedApp />
    </BrowserRouter>
  );;
};

const runAuthenticatedAppMocks = (
  getEnvConfigData: object,
  checkUserAuthenticationData: object,
  useUserAuthData: object,
  useErrorModalsData: object
): void => {
  mockGetEnvConfig.mockReturnValue(getEnvConfigData as unknown as ConfigType);
  mockCheckUserAuthentication.mockReturnValue(
    checkUserAuthenticationData as unknown as ReturnType<
      typeof checkUserAuthentication
    >
  );
  mockUseUserAuth.mockReturnValue(useUserAuthData as unknown as ReturnType<
    typeof useUserAuth
  >);
  mockUseDispatch.mockReturnValue(jest.fn() as unknown as ReturnType<
    typeof useDispatch
  >);
  mockUseErrorModals.mockReturnValue(useErrorModalsData as unknown as ReturnType<
    typeof useErrorModals
  >);
};
