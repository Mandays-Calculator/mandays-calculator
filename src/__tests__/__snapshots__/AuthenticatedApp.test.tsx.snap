// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN AuthenticatedApp Component is called WHEN page is still in verification process, THEN it should render the page loader: auth-page-loader 1`] = `
<div>
  <div
    class="MuiBox-root css-1g5urev"
  >
    <span
      aria-label="loading..."
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
    <h6
      class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
    >
      common.userManagement.auth.signInLoading
    </h6>
  </div>
</div>
`;

exports[`GIVEN AuthenticatedApp Component is called WHEN user is authenticated, THEN it user should be redirected to the landing page: user-authenticated 1`] = `
<div>
  <div
    class="css-1ezlr65"
    id="layout-container"
  >
    <div
      class="MuiBox-root css-k008qs"
    >
      <div
        class="MuiDrawer-root MuiDrawer-docked css-j1fxea-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
        >
          <ul
            class="MuiList-root MuiList-padding css-gygkfk-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1z0797q-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-19vlph4-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-18t382g-MuiSvgIcon-root"
                  data-testid="svg-component"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-icon_sample"
                  />
                </svg>
              </div>
            </li>
          </ul>
          <div
            class="css-4w50jn"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
                data-testid="svg-component"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <use
                  href="#icon-collapse_right"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-lo1nes"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-lr6u4k-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-disableGutters css-idqxm-MuiContainer-root"
        >
          <div
            class="css-vgodjd"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1hq4vw-MuiToolbar-root"
            >
              <div
                class="MuiBox-root css-lr0ish"
              >
                <div
                  aria-label="Open settings"
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="undefined undefined"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="[object Object]"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-p8vqrd-MuiTypography-root"
                    >
                      undefined undefined
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-11ine16-MuiTypography-root"
                    />
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
                  >
                    <button
                      aria-controls="menu-appbar"
                      aria-haspopup="true"
                      aria-label="account of current user"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-nff64r-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m7 10 5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="css-18gt0qv"
      />
    </div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-19bwc11-MuiPaper-root-MuiAppBar-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-184l77m-MuiTypography-root"
      >
        Â© 
        2024
         
        footer.label
      </p>
    </header>
  </div>
</div>
`;

exports[`GIVEN AuthenticatedApp Component is called WHEN user is not authenticated, THEN it user cannot proceed to the landing page: user-not-authenticated 1`] = `
<div>
  <div
    class="css-1ezlr65"
    id="layout-container"
  >
    <div
      class="MuiBox-root css-k008qs"
    >
      <div
        class="MuiDrawer-root MuiDrawer-docked css-j1fxea-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
        >
          <ul
            class="MuiList-root MuiList-padding css-gygkfk-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1z0797q-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-19vlph4-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-18t382g-MuiSvgIcon-root"
                  data-testid="svg-component"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <use
                    href="#icon-icon_sample"
                  />
                </svg>
              </div>
            </li>
          </ul>
          <div
            class="css-4w50jn"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3bwekg-MuiSvgIcon-root"
                data-testid="svg-component"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <use
                  href="#icon-collapse_right"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-lo1nes"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-lr6u4k-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiContainer-root MuiContainer-disableGutters css-idqxm-MuiContainer-root"
        >
          <div
            class="css-vgodjd"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1hq4vw-MuiToolbar-root"
            >
              <div
                class="MuiBox-root css-lr0ish"
              >
                <div
                  aria-label="Open settings"
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-7ldjcq-MuiGrid-root"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="undefined undefined"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="[object Object]"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-p8vqrd-MuiTypography-root"
                    >
                      undefined undefined
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-11ine16-MuiTypography-root"
                    />
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
                  >
                    <button
                      aria-controls="menu-appbar"
                      aria-haspopup="true"
                      aria-label="account of current user"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-nff64r-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m7 10 5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="css-18gt0qv"
      />
    </div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-19bwc11-MuiPaper-root-MuiAppBar-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-184l77m-MuiTypography-root"
      >
        Â© 
        2024
         
        footer.label
      </p>
    </header>
  </div>
</div>
`;
